### DISCUSSION TITLE
paper on GATv2 using Edge embedding

### DISCUSSION THREAD

**Discussion by XIANG64Young** [pyg-team/pytorch_geometric#7225] [Q&A]
Hi, I would like to know more details about the use of edge embedding for GATv2.
Just as [How Attentive are Graph Attention Networks](https://arxiv.org/abs/2105.14491) demonstrated the change from GAT to GATv2, [Edge-Featured Graph Attention Network](https://link.springer.com/chapter/10.1007/978-3-030-86362-3_21) demonstrates the effectiveness of GAT using edge embeddings.
However, I found no relevant articles proving the validity of using edge embedding on GATv2. If anyone finds it, please get back to me in time.

**Comment by rusty1s** [comment]
What do you mean with validity of using edge embeddings? Can you clarify?

**Reply by XIANG64Young** [reply]
In [How Attentive are Graph Attention Networks](https://arxiv.org/abs/2105.14491), the formula is $e(h_i, h_j)= \mathbf{a}^{\top}\mathrm{LeakyReLU}(W [h_i \, \Vert \, h_i])$.
But in [docs](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.nn.conv.GATv2Conv.html#torch_geometric.nn.conv.GATv2Conv), there exist a new formula using edge embeddings: $e(h_i, h_j, e_{ij})= \mathbf{a}^{\top}\mathrm{LeakyReLU}(W [h_i \, \Vert \, h_i ,\Vert \ e_{ij}])$.
I wondered if there was some paper that demonstrated the validity of the new formula in some experiments or mathematics, just like [Edge-Featured Graph Attention Network](https://link.springer.com/chapter/10.1007/978-3-030-86362-3_21) on GAT.

**Reply by rusty1s** [reply]
Actually no, this was just added by us based on popular request.

**Reply by XIANG64Young** [reply]
Well.  Thank you for your generous response!
